op-acceptance-tests/tests/interop/seqwindow/expiry_test.go

```diff
@@ -0,0 +1,104 @@
+	seqWindowSize := sys.L2ChainA.Escape().RollupConfig().SeqWindowSize
+	estimatedExpiryNum := last.Source.Number + seqWindowSize
+	lastRef, err := sys.L1EL.Escape().EthClient().BlockRefByHash(t.Ctx(), last.Source.Hash)
+	require.NoError(err)
+	lastTime := time.Unix(int64(lastRef.Time), 0)
+	l1BlockTime := sys.L1EL.EstimateBlockTime()
+	windowDuration := l1BlockTime * time.Duration(seqWindowSize)
+	t.Logger().Info("Sequencing window expiry",
+		"estimateL1Num", estimatedExpiryNum, "windowDuration", windowDuration,
+		"fromNow", lastTime.Add(windowDuration).Sub(time.Now()))
+
+	// The unsafe L2 block after this last safe block is going to be reorged out
+	// once the sequencing window expires.
+	// However, since it is empty, it may stay around, because it would be compatible.
+	// So let's insert a transaction, then we can be sure it is different.
+	tx2 := alice.Transfer(common.HexToAddress("0xdead"), eth.GWei(42))
+	receipt2, err := tx2.Included.Eval(t.Ctx())
+	require.NoError(err)
+	// Now get the block that included the tx. This block will change.
+	old := eth.BlockID{Hash: receipt2.BlockHash, Number: receipt2.BlockNumber.Uint64()}
+	t.Logger().Info("Confirmed tx 2, which will be reorged out later",
+		"tx", receipt2.TxHash, "l2Block", old)
+	// The logs will show a "Chain reorg detected" from op-geth.
+
+	t.Logger().Info("Waiting for sequencing window expiry induced reorg now")
+	// Monitor that the old unsafe chain is reorged out as expected
+	require.Eventually(func() bool {
+		latest := sys.L2ELA.BlockRefByNumber(old.Number)
+		return latest.Hash != old.Hash
+	}, windowDuration+time.Second*20, time.Second, "expecting old block to be reorged out")
+
+	t.Logger().Info("Waiting for supervisor to surpass pre-reorg chain now")
+	// Monitor that the supervisor can continue to sync.
+	// A lot more blocks will expire first; the local-safe chain will be entirely force-derived blocks.
+	require.Eventually(func() bool {
+		safe, err := sys.Supervisor.Escape().QueryAPI().CrossSafe(t.Ctx(), sys.L2ChainA.ChainID())
+		require.NoError(err)
+		return safe.Source.Number > estimatedExpiryNum
+	}, windowDuration+time.Second*20, time.Second, "expecting supervisor to sync cross-safe data, after resolving sequencing window expiry")
+
+	t.Logger().Info("Sanity-checking now")
+	// Sanity-check the unsafe head of the supervisor is also updated
+	tip, err := sys.Supervisor.Escape().QueryAPI().LocalUnsafe(t.Ctx(), sys.L2ChainA.ChainID())
+	require.NoError(err)
+	require.True(tip.Number > estimatedExpiryNum)
+	// Sanity-check the supervisor is on the right chain
+	safe, err := sys.Supervisor.Escape().QueryAPI().CrossSafe(t.Ctx(), sys.L2ChainA.ChainID())
+	require.NoError(err)
+	other := sys.L2ELA.BlockRefByNumber(safe.Derived.Number)
+	require.Equal(safe.Derived.Hash, other.Hash, "supervisor must match chain with EL")
+
+	// re-enable the batcher now that we are done with the test.
+	t.Require().NoError(sys.L2BatcherA.ActivityAPI().StartBatcher(t.Ctx()))
```

Comment:
Is it worth submitting another L2 tx after the sequencer is deriving again to make sure we really do fully recover and start including new tx and getting them into batches so they become safe (rather than them being re-orged out again because the sequencer window keeps expiring)?

Reply:
I think there's a rabbit-hole there with auto-enabling sequencer recovery mode. But we can do that, or make the test enable it via RPC. I'll add a TODO in the main issue. I've split out the test as separate issue.

Reply:
Ah, I don't want to create rabbit holes - it seemed like a relatively quick win on the surface.  Probably is worth considering as a future improvement but don't take it as a blocker unless you feel its necessary to give us sufficient confidence that we can handle a sequencer window expiry.